junifer/storage/tests/test_hdf5.py

```diff
@@ -0,0 +1,980 @@
+        storage._store_data(
+            kind="vector",
+            meta_md5="md5",
+            element=[{"sub": "001"}],
+            data=np.empty((1, 1)),
+            col_names="col",
+        )
+
+
+@pytest.mark.parametrize(
+    "force, dtype",
+    [
+        (True, "float32"),
+        (False, "float64"),
+    ],
+)
+def test_f64_to_f632_conversion(
+    tmp_path: Path, force: bool, dtype: str
+) -> None:
+    """Test actual data is casted from float64 to float32.
+
+    Parameters
+    ----------
+    tmp_path : pathlib.Path
+        The path to the test directory.
+    force : bool
+        The parametrized conversion option.
+    dtype : str
+        The parametrized expected data type.
+
+    """
+    uri = tmp_path / "test_data_conversion.hdf5"
+    storage = HDF5FeatureStorage(uri=uri, force_float32=force)
+    # Metadata to store
+    meta = {
+        "element": {"subject": "test"},
+        "dependencies": ["numpy"],
+        "marker": {"name": "mark"},
+        "type": "BOLD",
+    }
+    # Process the metadata
+    meta_md5, meta_to_store, element_to_store = process_meta(meta)
+    # Store matrix
+    storage.store_metadata(
+        meta_md5=meta_md5, element=element_to_store, meta=meta_to_store
+    )
+    # Store data
+    storage.store_matrix(
+        meta_md5=meta_md5,
+        element=element_to_store,
+        data=np.arange(4, dtype="float64").reshape((2, 2)),
+    )
+    # Read into dataframe
+    read_df = storage.read_df(feature_md5=meta_md5)
+    # Check data type
+    assert read_df.values.dtype == np.dtype(dtype)
+
+
+def test_store_matrix(tmp_path: Path) -> None:
+    """Test matrix store.
+
+    Parameters
+    ----------
+    tmp_path : pathlib.Path
+        The path to the test directory.
+
+    """
+    uri = tmp_path / "test_store_matrix.hdf5"
+    storage = HDF5FeatureStorage(uri=uri)
+    # Metadata to store
+    meta = {
+        "element": {"subject": "test"},
+        "dependencies": ["numpy"],
+        "marker": {"name": "fc"},
+        "type": "BOLD",
+    }
+    # Process the metadata
+    meta_md5, meta_to_store, element_to_store = process_meta(meta)
+    # Store metadata
+    storage.store_metadata(
+        meta_md5=meta_md5, element=element_to_store, meta=meta_to_store
+    )
+
+    # Store 4 X 3 full matrix
+    data = np.array(
+        [[1, 2, 3], [11, 22, 33], [111, 222, 333], [1111, 2222, 3333]]
+    )
+    row_headers = ["row1", "row2", "row3", "row4"]
+    col_headers = ["col1", "col2", "col3"]
+
+    # Store matrix
+    storage.store_matrix(
+        meta_md5=meta_md5,
+        element=element_to_store,
+        data=data,
+        row_names=row_headers,
+        col_names=col_headers,
+    )
+
+    # List the stored features
+    features = storage.list_features()
+    # Check the MD5
+    assert "BOLD_fc" == features[meta_md5]["name"]
+
+    # Read into dataframe
+    read_df = storage.read_df(feature_md5=meta_md5)
+    # Check shape of dataframe
+    assert read_df.shape == (4, 3)
```

Comment:
See my comment on `store_matrix`. We should have one row per sample. In junifer terms, one row per element.

Reply:
Addressed.